File: 459-repeated-substring-pattern/repeated-substring-pattern.py
import re

class Solution:
    def repeatedSubstringPattern(self, s: str) -> bool:
        if len(s) > 1:
            if len(set(s)) == 1:
                return True
            for i in range(2, len(s)//2+1):
                if len(s) % i == 0 and len(s) // i == (len(match) if (match := re.findall(s[:i], s)) else -1):
                    return True
        return False

File: 459-repeated-substring-pattern/repeated-substring-pattern.java
class Solution {
    public boolean repeatedSubstringPattern(String s) {
        boolean answer = false;
        for (int i = 1; i <= s.length() / 2; i++) {
            if (s.length() % i == 0) {
                answer = String.join("", s.split(s.substring(0, i))).length() == 0;
                if (answer) break;
            }
        }
        return answer;        
    }
}